public_html/src/MyGame/DyePack.js

```diff
@@ -0,0 +1,80 @@
+/* 
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+"use strict";
+
+function DyePack(sprite) {
```

Comment:
Could you use ES6? It allows for actual classes in JS. [ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes)
It would look something like this:
```suggestion
class DyePack {
   constructor(sprite) {
     this.kSprite = sprite;
     ...
     this.mCurSpeed = 0;
   }
   
   initialize(xPos, yPos) { ... }
   hitDyePack() { ... }
   getIsTerminated() { ... }
   ...
}
```